### PR TITLE
Allow assigning groups in staff mutations

### PR DESCRIPTION
**Pull request by IKarbowiak**
Allow assigning groups in staff create mutations. Allow assigning and unassigning groups in staff create mutations.

# Impact

* [ ] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [x] Database queries are optimized and the number of queries is constant
* [x] Database migration files are up to date
* [x] The changes are tested
* [x] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/account/mutations/staff.py [hunk 1]
```diff
@@ -156,30 +180,74 @@ class Meta:
...
 
+        errors = defaultdict(list)
         if cleaned_input.get("redirect_url"):
             try:
                 validate_storefront_url(cleaned_input.get("redirect_url"))
             except ValidationError as error:
-                raise ValidationError(
-                    {"redirect_url": error}, code=AccountErrorCode.INVALID
-                )
+                error.code = AccountErrorCode.INVALID
```
Comment (maarcingebala) [inline saleor/graphql/account/mutations/staff.py]: Not sure if we shouldn't introduce `StaffError` enum here? I guess we should discuss that with the team.

File: tests/api/test_account.py [hunk 2]
```diff
@@ -1617,48 +1687,259 @@ def test_staff_update(staff_api_client, permission_manage_staff, media_root):
...
+    permission_manage_staff,
+    media_root,
+    permission_manage_orders,
+    permission_group_manage_users,
+    permission_manage_products,
+    permission_manage_users,
+):
+    query = STAFF_UPDATE_MUTATIONS
+
+    group = permission_group_manage_users
```
Comment (fowczarek) [inline tests/api/test_account.py:1898]: Maby better use `bulk_create` ?

File: tests/api/test_account.py [hunk 3]
```diff
@@ -1617,48 +1687,259 @@ def test_staff_update(staff_api_client, permission_manage_staff, media_root):
...
+    group2 = Group.objects.create(name="manage orders")
+    group2.permissions.add(permission_manage_orders)
+    group3 = Group.objects.create(name="manage products")
+    group3.permissions.add()
```
Comment (fowczarek) [inline tests/api/test_account.py:1902]: What is a reason to add nothing?

File: saleor/graphql/account/mutations/staff.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/account/mutations/staff.py:195]: If we agree that it's a good idea then let's change it (I see two thumbs-ups 😉).

File: tests/api/test_account.py [hunk 4]
```diff
@@ -1617,48 +1687,259 @@ def test_staff_update(staff_api_client, permission_manage_staff, media_root):
...
+    staff_api_client,
+    media_root,
+    permission_group_manage_users,
+    permission_manage_staff,
+    permission_manage_users,
+    permission_manage_orders,
+    permission_manage_products,
+):
+    query = STAFF_UPDATE_MUTATIONS
+    group1 = permission_group_manage_users
```
Comment (fowczarek) [inline tests/api/test_account.py:1728]: Maby better use `bulk_create`?

File: tests/api/test_account.py [hunk 5]
```diff
@@ -1617,48 +1687,259 @@ def test_staff_update(staff_api_client, permission_manage_staff, media_root):
...
+    staff_api_client,
+    permission_manage_staff,
+    media_root,
+    permission_manage_orders,
+    permission_group_manage_users,
+    permission_manage_products,
+):
+    query = STAFF_UPDATE_MUTATIONS
+
+    group = permission_group_manage_users
```
Comment (fowczarek) [inline tests/api/test_account.py:1838]: Maby better use `bulk_create`?

File: tests/api/test_account.py (hunk 3 above)
Comment (IKarbowiak) [reply tests/api/test_account.py:1902]: No reason. I forgot to remove 😄

Comment (fowczarek) [reply tests/api/test_account.py:1902]: :police_car:

File: saleor/graphql/account/mutations/staff.py (hunk 1 above)
Comment (IKarbowiak) [reply saleor/graphql/account/mutations/staff.py:195]: After the team discussion, we probably should leave it as it is.

**Pull request merged**